specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json

```diff
@@ -0,0 +1,794 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Insights API",
+    "version": "2018-04-16",
+    "x-ms-code-generation-settings": {
+      "name": "MonitorManagementClient"
+    },
+    "description": "Azure Monitor client to create/update/delete Scheduled Query Rules"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/microsoft.insights/scheduledQueryRules/{ruleName}": {
+      "put": {
+        "description": "Creates or updates an log search rule.\r\nRequest method: PUT\t\tRequest URI: https://management.azure.com/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/microsoft.insights/scheduledQueryRules/{logsearch-rule-name}?api-version={api-version}",
+        "tags": [
+          "scheduledQueryRules"
+        ],
+        "operationId": "createOrUpdateScheduledQueryRules",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/LogSearchRuleResource"
+            },
+            "description": "The parameters of the rule to create or update."
+          }
+        ],
+        "responses": {
```

Comment:
You should support a `"default"` response for errors. (see example https://github.com/abversqr/azure-rest-api-specs/blob/2e4c790d16eb6be9299db7a75798bca1697fd5e9/specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json#L503 )